ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -510,6 +510,42 @@ public Container getContainerById(short accountId, Short containerId) throws Acc
     return container;
   }
 
+  @Override
+  public void addDataset(Dataset dataset) throws AccountServiceException {
+    try {
+      String accountName = dataset.getAccountName();
+      String containerName = dataset.getContainerName();
+      Container container = getContainerByName(accountName, containerName);
+      if (container == null) {
+        throw new AccountServiceException("Can't find the container: " + containerName + " in account: " + accountName,
+            AccountServiceErrorCode.BadRequest);
+      }
+      short accountId = container.getParentAccountId();
+      short containerId = container.getId();
+      mySqlAccountStore.addDataset(accountId, containerId, dataset);
```

Comment:
So the dataset update is a simple replacement?

Reply:
I'll have another pr to support update dataset.